dashboard/modules/metrics/metrics_head.py

```diff
@@ -23,14 +31,76 @@
 )
 
 
+class TaskProgress(BaseModel):
+    num_finished: int = 0
+    num_running: int = 0
+    nun_scheduled: int = 0
+    num_waiting_for_dependencies: int = 0
+    num_waiting_for_execution: int = 0
+
+
+prometheus_metric_map = {
+    "FINISHED": "num_finished",
+    "RUNNING": "num_running",
+    "SCHEDULED": "nun_scheduled",
+    "WAITING_FOR_DEPENDENCIES": "num_waiting_for_dependencies",
+    "WAITING_FOR_EXECUTION": "num_waiting_for_execution",
```

Comment:
This has been updated in master; I think the new mapping should be:

```
    "FINISHED": "num_finished",
    "PENDING_ARGS_AVAIL": "num_pending_args_avail",
    "SUBMITTED_TO_WORKER": "num_submitted_to_worker"

    # Group together.
    "RUNNING": "num_running",
    "RUNNING_IN_RAY_GET": "num_running",
    "RUNNING_IN_RAY_WAIT": "num_running",

    # Group together
    "PENDING_NODE_ASSIGNMENT": "num_pending_node_assignment",
    "PENDING_ARGS_FETCH": "num_pending_node_assignment",
    "PENDING_OBJ_STORE_MEM_AVAIL": "num_pending_node_assignment",

 ```

Reply:
There's a enum defined in `custom_types.py` under `ray.experimental.state`. Maybe we can reuse this?

Reply:
@rkooo567 , that enum is just strings right? how would you suggest I re-use it?

Reply:
What's the progression of the states?

num_pending_args_avail  -> num_pending_node_assignment -> num_submitted_to_worker -> num_running -> num_finished?

Reply:
PENDING_ARGS_AVAIL -> SUBMITTED_TO_WORKER -> PENDING_NODE_ASSIGNMENT/PENDING_ARGS_FETCH/PENDING_NODE_ASSIGNMENT -> RUNNING/RUNNING_IN_RAY_GET/RUNNING_IN_RAY_WAIT -> FINISHEd

Reply:
> @rkooo567 , that enum is just strings right? how would you suggest I re-use it?

Ah, I didn't know we want to map state -> num_*. (I just meant to use the enum to obtain all available states). Should we need this mapping? Is it too verbose if we render all states from the progress bar?

Reply:
The order is actually

PENDING_ARGS_AVAIL -> PENDING_NODE_ASSIGNMENT -> SUBMITTED_TO_WORKER -> RUNNING

(per the order in common.proto).

For the progress bar, actually how about we simplify into three stages?
- Waiting for dependencies (pending args)
- Waiting for scheduling (everything before running)
- Running

We can add further breakdowns in the future with tooltips, etc. I guess users will probably be confused with more breakdowns and should refer to the task metric view for that.